src/tests/Microsoft.VisualStudio.Composition.Tests/MetadataViewClassDefaultCtorProviderTests.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.VisualStudio.Composition.Tests
+{
+    using System;
+    using System.ComponentModel;
+    using System.Composition;
+    using Xunit;
+
+    public class MetadataViewClassDefaultCtorProviderTests
+    {
+        private const string PublicPropertyExpectedValue = "Public Property Value";
+
+        private const string InternalPropertyExpectedValue = "Internal Property Value";
+
+        private const string DefaultPropertyExpectedValue = "Some default value";
+
+        [MefFact(CompositionEngines.V2Compat, typeof(MetadataDecoratedPart), typeof(DefaultMetadataViewImporter))]
+        public void PublicCtorPublicProperty(IContainer container)
+        {
+            var part = container.GetExportedValue<DefaultMetadataViewImporter>();
+            Assert.Equal(PublicPropertyExpectedValue, part.InnerPart.Metadata.PublicProperty);
+        }
+
+        [MefFact(CompositionEngines.V2, typeof(MetadataDecoratedPart), typeof(DerivedMetadataViewImporter), NoCompatGoal = true)]
+        public void PropertyWithPrivateSetterInBaseClass_V2(IContainer container)
+        {
+            var part = container.GetExportedValue<DerivedMetadataViewImporter>();
+            Assert.Null(part.InnerPart.Metadata.PropertyWithPrivateSetter);
+        }
+
+        [MefFact(CompositionEngines.V3EmulatingV2, typeof(MetadataDecoratedPart), typeof(DerivedMetadataViewImporter))]
+        public void PropertyWithPrivateSetterInBaseClass_V3(IContainer container)
+        {
+            var part = container.GetExportedValue<DerivedMetadataViewImporter>();
+            Assert.Equal(PublicPropertyExpectedValue, part.InnerPart.Metadata.PropertyWithPrivateSetter);
+        }
+
+        [MefFact(CompositionEngines.V2Compat, typeof(MetadataDecoratedPart), typeof(DefaultMetadataViewImporter))]
+        public void DefaultProperty(IContainer container)
+        {
+            var part = container.GetExportedValue<DefaultMetadataViewImporter>();
+            Assert.Equal(DefaultPropertyExpectedValue, part.InnerPart.Metadata.PropertyWithDefault);
+        }
+
+        [MefFact(CompositionEngines.V2Compat, typeof(MetadataDecoratedPart), typeof(DefaultMetadataViewImporter))]
+        public void PublicCtorInternalProperty(IContainer container)
+        {
+            var part = container.GetExportedValue<DefaultMetadataViewImporter>();
+            Assert.Null(part.InnerPart.Metadata.InternalProperty);
+        }
+
+        [MefFact(CompositionEngines.V2Compat, typeof(MetadataDecoratedPart), typeof(InternalCtorMetadataViewImporter), InvalidConfiguration = true)]
+        public void InternalCtorPublicProperty(IContainer container)
+        {
+            var part = container.GetExportedValue<InternalCtorMetadataViewImporter>();
```

Comment:
Is this test testing something?

Reply:
Yes, although I can see how it's not clear. In the `MefFact` attribute on this test method, there is this property set: `InvalidConfiguration = true`. This asserts that either during composition or during test method execution that a MEF exception is thrown.